tests/train_test_split/test_base.py

```diff
@@ -0,0 +1,238 @@
+from typing import Union
+
+from quantfinlib.train_test_split._base import BaseCV, _purge, _embargo
+from sklearn.model_selection import KFold
+
+import numpy as np
+import pandas as pd
+import pytest
+
+
+def _remove_overlapping_groups(
+    train_index: np.ndarray, test_index: np.ndarray, indices: np.ndarray, groups: Union[np.ndarray, pd.DatetimeIndex]
+) -> np.ndarray:
+    """Remove train indices whose groups overlap between the training and test sets and add them to the test set.
+
+    Parameters
+    ----------
+    train_index : np.ndarray
+        The initial training indices.
+    test_index : np.ndarray
+        The test indices.
+    indices : np.ndarray
+        All indices.
+    groups : Union[np.ndarray, pd.DatetimeIndex]
+        The groups.
+
+    Returns
+    -------
+    np.ndarray
+        The updated training indices.
+    """
+    train_groups, test_groups = groups[train_index], groups[test_index]
+    overlapping_groups = np.intersect1d(train_groups, test_groups)
+    overlapping_index = indices[np.isin(groups, overlapping_groups)]
+    train_index = np.sort(np.setdiff1d(train_index, overlapping_index))
+    test_index = np.sort(np.union1d(test_index, overlapping_index))
+    return train_index, test_index
+
+
+def integer_even_groups():
+    return np.hstack([np.zeros(10) + i for i in range(10)])
+
+
+def integer_uneven_groups():
+    groups = np.random.randint(0, 33, 100)
+    # make sure that all groups between 0 and 32 are present in the data
+    groups = np.unique(groups, return_inverse=True)[1].reshape(groups.shape)
+    groups.sort()
+    return groups
+
+
+def datetime_even_groups():
+    return pd.date_range("2020-01-01", periods=100, freq="D")
+
+
+def datetime_uneven_groups():
+    groups = pd.date_range("2020-01-01", periods=100, freq="D")
+    group_indices = np.random.randint(0, 60, 100)
+    # make sure that all groups between 0 and 32 are present in the data
+    group_indices = np.unique(group_indices, return_inverse=True)[1].reshape(group_indices.shape)
+    group_indices.sort()
+    return groups[group_indices]
+
+
+@pytest.fixture(
+    params=[
+        ("integer", np.arange(100), integer_even_groups()),

```

Comment:
I wanted to run tests with profiling on larger arrays to see where are the processing bottlenecks, but it is not so easy as there are many places where 100 needs to be changed to million or smth. can we define some variables on top to easily control testing parameters? I understand, this is probably not the point of unit-tests...

Reply:
I improved this in the unittests.